functions/event_handler/main.py

```diff
@@ -55,12 +69,139 @@ def store_pub_sub_event_in_bigquery(cloud_event):
         "backend_metadata": backend_metadata,
     }
 
-    logger.info("Attempting to store row: %r.", row)
+    logger.info("Attempting to store event: %r.", row)
 
-    client = Client()
-    errors = client.insert_rows(table=client.get_table(BIGQUERY_EVENTS_TABLE), rows=[row])
+    bigquery_client = BigQueryClient()
+    errors = bigquery_client.insert_rows(table=bigquery_client.get_table(bigquery_events_table), rows=[row])
 
     if errors:
         raise ValueError(errors)
 
-    logger.info("Successfully stored row in %r.", BIGQUERY_EVENTS_TABLE)
+    logger.info("Successfully stored event in %r.", bigquery_events_table)
+
+    if original_event["kind"] not in {"question", "cancellation"}:
+        return
+
+    _authenticate_with_kubernetes_cluster()
+    batch_api = kubernetes.client.BatchV1Api()
+
+    if original_event["kind"] == "question":
+        _dispatch_question_as_kueue_job(original_event, original_attributes, batch_api)
+    elif original_event["kind"] == "cancellation":
+        _cancel_kueue_job(original_attributes["question_uuid"], batch_api)
+
+
+def _dispatch_question_as_kueue_job(event, attributes, batch_api):
+    """Dispatch a question event to Kueue as a job.
+
+    :param dict event: a question event from an Octue Twined service
+    :param dict attributes: the attributes accompanying the question event
+    :param kubernetes.client.BatchV1Api batch_api: the kubernetes batch API
+    :return None:
+    """
+    octue_services_topic_name = os.environ["OCTUE_SERVICES_TOPIC_NAME"]
+    kubernetes_service_account_name = os.environ["KUBERNETES_SERVICE_ACCOUNT_NAME"]
+    kueue_local_queue = os.environ["KUEUE_LOCAL_QUEUE"]
+    artifact_registry_repository_url = os.environ["ARTIFACT_REGISTRY_REPOSITORY_URL"]
+    question_default_cpus = int(os.environ["QUESTION_DEFAULT_CPUS"])
+    question_default_memory = os.environ["QUESTION_DEFAULT_MEMORY"]
+    question_default_ephemeral_storage = os.environ["QUESTION_DEFAULT_EPHEMERAL_STORAGE"]
```

Comment:
How is it that these defaults are set by the event handler? Are they defaults for all services in the infrastructure then? I think we had this discussion but I didn't quite realise it was here that they'd be defined (I thought it was somewhere per-service).

It's OK for now but we probably want to figure out a better way of setting service defaults.

Reply:
Tracked in this issue: https://github.com/octue/octue-sdk-python/issues/726